fs/fs.go

```diff
@@ -95,13 +102,12 @@ func NewFilesystem(root string, cfg config.Config, opts ...Option) (_ snapshot.F
 		return nil, errors.Wrapf(err, "failed to setup resolver")
 	}
 
-	var ns *metrics.Namespace
-	if !cfg.NoPrometheus {
+	nsLock.Lock()
+	defer nsLock.Unlock()
+	if !cfg.NoPrometheus && ns == nil {
 		ns = metrics.NewNamespace("stargz", "fs", nil)
-	}
-	c := fsmetrics.NewLayerMetrics(ns)
-	if ns != nil {
 		metrics.Register(ns)
+		metricsCtr = fsmetrics.NewLayerMetrics(ns)
 	}
```

Comment:
`fs.metricsController` will be `nil` if `cfg.NoPrometheus` is `true`. 
Doesn't it cause panic on `fs.metricsController.Add()` during `fs.Mount`?

Reply:
Fixed